extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcSessionImpl.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.oidc.runtime;
+
+import java.time.Instant;
+import java.util.function.Function;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+import io.quarkus.oidc.IdToken;
+import io.quarkus.oidc.OidcSession;
+import io.quarkus.oidc.OidcTenantConfig;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class OidcSessionImpl implements OidcSession {
+
+    @Inject
+    RoutingContext routingContext;
+
+    @Inject
+    DefaultTenantConfigResolver resolver;
+
+    @Inject
+    @IdToken
+    JsonWebToken idToken;
+
+    @Override
+    public String getTenantId() {
+        return routingContext.get(OidcUtils.TENANT_ID_ATTRIBUTE);
+    }
+
+    @Override
+    public Uni<Void> logout() {
+        String sessionCookieName = routingContext.get(OidcUtils.SESSION_COOKIE_NAME);
+        if (sessionCookieName != null) {
+            Uni<OidcTenantConfig> oidcConfigUni = resolver.resolveConfig(routingContext);
+            return oidcConfigUni.onItem().transformToUni(new Function<OidcTenantConfig, Uni<? extends Void>>() {
```

Comment:
I have a feeling that if we get to the point where the user wants to log out, we already should have resolved the `OidcTenantConfig`, no? We should just store that once we resolve it, no? It can't possibly change during a request, can it?

Reply:
@FroMage I agree, I actually typed to wait indefinitely with the comment it is already resolved. That said, Clement is not happy when he sees `await.indefinitely()` :-) so I replaced it, the resolved config is stored internally in the map (it can be dynamic so `namedTenants` is not enough)